src/components/common/deleteResource.jsx

```diff
@@ -97,14 +97,16 @@ DeleteResourceModal.propTypes = {
     onClose: PropTypes.func.isRequired,
 };
 
-export const DeleteResourceButton = ({ objectId, disabled, overlayText, actionName, showDialog }) => {
+export const DeleteResourceButton = ({ objectId, disabled, overlayText, actionName, showDialog, variant, isInline }) => {
     if (disabled) {
         return (
             <Tooltip id={`delete-${objectId}-tooltip`}
                      content={overlayText}>
                 <span>
                     <Button id={`delete-${objectId}`}
-                        variant='danger'
+                        variant={variant || 'danger'}
+                        isDanger={!!variant}
+                        isInline={isInline}
```

Comment:
DRY. Please create the `<Button>` once with `isDisabled={disabled}`, and wrap it into the `<Tooltip>` conditionally.